File: projetws/app/src/main/java/com/example/projetws/beans/Etudiant.java
package com.example.projetws.beans;

import com.google.gson.annotations.SerializedName;

public class Etudiant {
    private int id;
    private String nom;
    private String prenom;
    private String ville;
    private String sexe;
    @SerializedName("date_naissance")
    private String dateNaissance; // New field for date of birth
    @SerializedName("image")
    private String imagePath; // New field for storing image path

    public Etudiant(int id, String nom, String prenom, String ville, String sexe, String dateNaissance, String imagePath) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.ville = ville;
        this.sexe = sexe;
        this.dateNaissance = dateNaissance;
        this.imagePath = imagePath;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getSexe() {
        return sexe;
    }

    public void setSexe(String sexe) {
        this.sexe = sexe;
    }

    public String getDateNaissance() {
        return dateNaissance;
    }

    public void setDateNaissance(String dateNaissance) {
        this.dateNaissance = dateNaissance;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }

    @Override
    public String toString() {
        return "Etudiant{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prenom='" + prenom + '\'' +
                ", ville='" + ville + '\'' +
                ", sexe='" + sexe + '\'' +
                ", dateNaissance='" + dateNaissance + '\'' +
                ", imagePath='" + imagePath + '\'' +
                '}';
    }
}

File: projet/Source Files/classes/Etudiant.php
<?php

class Etudiant
{
    private $id;
    private $nom;
    private $prenom;
    private $ville;
    private $sexe;
    private $dateNaissance; // Nouveau champ
    private $image; // Nouveau champ

    function __construct($id, $nom, $prenom, $ville, $sexe, $dateNaissance = null, $image = null)
    {
        $this->id = $id;
        $this->nom = $nom;
        $this->prenom = $prenom;
        $this->ville = $ville;
        $this->sexe = $sexe;
        $this->dateNaissance = $dateNaissance;
        $this->image = $image;
    }

    function getId()
    {
        return $this->id;
    }

    function getNom()
    {
        return $this->nom;
    }

    function getPrenom()
    {
        return $this->prenom;
    }

    function getVille()
    {
        return $this->ville;
    }

    function getSexe()
    {
        return $this->sexe;
    }

    function getDateNaissance()
    {
        return $this->dateNaissance;
    }

    function getImage()
    {
        return $this->image;
    }

    function setId($id)
    {
        $this->id = $id;
    }

    function setNom($nom)
    {
        $this->nom = $nom;
    }

    function setPrenom($prenom)
    {
        $this->prenom = $prenom;
    }

    function setVille($ville)
    {
        $this->ville = $ville;
    }

    function setSexe($sexe)
    {
        $this->sexe = $sexe;
    }

    function setDateNaissance($dateNaissance)
    {
        $this->dateNaissance = $dateNaissance;
    }

    function setImage($image)
    {
        $this->image = $image;
    }

    public function __toString()
    {
        return $this->nom . " " . $this->prenom;
    }
}
